std/traits.d

```diff
@@ -5074,6 +5075,31 @@ template BuiltinTypeOf(T)
     else                                        static assert(0);
 }
 
+/++
+    Helper template so that `typeof can be used with templates taking other
+    templates (such as $(REF, staticMap, std, meta)).
```

Comment:
Plenty of other templates in this module use the word "returns", so please add a `Returns:` section here, as well as `Params:`. See https://github.com/jmdavis/phobos/blob/95112dc7cabe31ab3b63679d0c8a283d6dfef496/std/traits.d#L2034 for an example.